dali/kernels/audio/mel_scale/mel_filter_bank_gpu.cu

```diff
@@ -157,12 +157,12 @@ class MelFilterBankGpu<T>::Impl : public MelFilterImplBase<T> {
     } else {
       FillBlockDescsOuterFft(in_list, out_list);
     }
-    auto block_descs = scratchpad->ToGPU(stream, block_descs_);
-    auto interval_ends = scratchpad->ToGPU(stream, interval_ends_);
-    auto weights_down = scratchpad->ToGPU(stream, weights_down_);
-    T *norm_factors = nullptr;
-    if (args_.normalize)
-      norm_factors = scratchpad->ToGPU(stream, norm_factors_);
+    BlockDesc<T> *block_descs = nullptr;
+    int *interval_ends = nullptr;
+    T *weights_down = nullptr, *norm_factors = nullptr;
+    std::tie(block_descs, interval_ends, weights_down, norm_factors) =
+          scratchpad->ToContiguousGPU(stream, block_descs_, interval_ends_,
+                                      weights_down_, norm_factors_);
```

Comment:
What if `norm_factors_` is nullptr? Would `ToContiguousGPU` return nullptr as well?

Reply:
`norm_factors_` is a `vector` and `ToContiguousGPU` deals in collections, not pointers. This means that it will simply have nothing to copy from there. I don't think it would return `nullptr`, though (unless _all_ collections were empty).